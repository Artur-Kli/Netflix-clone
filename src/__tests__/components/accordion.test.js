import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import faqsData from '../../fixtures/faqs.json'
import { Accordion } from '../../components'

describe('<Accordion />', () => {
  it('render the <Accordion /> with populated data', () => {
    const {container, getByText} = render(
      <Accordion>
        <Accordion.Title>
          Frequently Asked Question
        </Accordion.Title>
        <Accordion.Frame>
          {faqsData.map(item =>
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          )}
        </Accordion.Frame>
      </Accordion>
    )

    expect(getByText('Frequently Asked Question')).toBeTruthy()
    expect(getByText('What is Netflix?')).toBeTruthy()
    expect(getByText('How much does Netflix cost?')).toBeTruthy()
    expect(getByText('Where can I watch?')).toBeTruthy()
    expect(getByText('How do I cancel?')).toBeTruthy()
    expect(getByText('What can I watch on Netflix?')).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('opens and closes the <Accordion /> component', () => {
    const {container, queryByText} = render(
      <Accordion>
        <Accordion.Title>
          Frequently Asked Question
        </Accordion.Title>
        <Accordion.Frame>
          {faqsData.map(item =>
            <Accordion.Item key={item.id}>
              <Accordion.Header>{item.header}</Accordion.Header>
              <Accordion.Body>{item.body}</Accordion.Body>
            </Accordion.Item>
          )}
        </Accordion.Frame>
      </Accordion>
    )

    const whatIsNetflixBodyText = "Watch Netflix on your smartphone, tablet, smart TV, laptop or streaming device, all for one low fixed monthly fee. Plans start from Â£5.99 a month. No extra costs or contracts."
      
    expect(queryByText(whatIsNetflixBodyText)).toBeFalsy()
    fireEvent.click(queryByText('How much does Netflix cost?'))
    expect(queryByText(whatIsNetflixBodyText)).toBeTruthy()

    fireEvent.click(queryByText('How much does Netflix cost?'))
    expect(queryByText(whatIsNetflixBodyText)).toBeFalsy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
