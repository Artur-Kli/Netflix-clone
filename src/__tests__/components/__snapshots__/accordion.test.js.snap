// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> opens and closes the <Accordion /> component 1`] = `
<div
  class="accordion__Container-l3iym9-0 fXsxPD"
>
  <div
    class="accordion__Inner-l3iym9-1 hhulRI"
  >
    <h1
      class="accordion__Title-l3iym9-3 bqdeAY"
    >
      Frequently Asked Question
    </h1>
    <div
      class="accordion__Frame-l3iym9-2 edcybJ"
    >
      <div
        class="accordion__Item-l3iym9-4 fLZyeo"
      >
        <div
          class="accordion__Header-l3iym9-5 kYfwMS"
        >
          What is Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-4 fLZyeo"
      >
        <div
          class="accordion__Header-l3iym9-5 kYfwMS"
        >
          How much does Netflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-4 fLZyeo"
      >
        <div
          class="accordion__Header-l3iym9-5 kYfwMS"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-4 fLZyeo"
      >
        <div
          class="accordion__Header-l3iym9-5 kYfwMS"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-4 fLZyeo"
      >
        <div
          class="accordion__Header-l3iym9-5 kYfwMS"
        >
          What can I watch on Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> render the <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-l3iym9-0 fXsxPD"
>
  <div
    class="accordion__Inner-l3iym9-1 hhulRI"
  >
    <h1
      class="accordion__Title-l3iym9-3 bqdeAY"
    >
      Frequently Asked Question
    </h1>
    <div
      class="accordion__Frame-l3iym9-2 edcybJ"
    >
      <div
        class="accordion__Item-l3iym9-4 fLZyeo"
      >
        <div
          class="accordion__Header-l3iym9-5 kYfwMS"
        >
          What is Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-4 fLZyeo"
      >
        <div
          class="accordion__Header-l3iym9-5 kYfwMS"
        >
          How much does Netflix cost?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-4 fLZyeo"
      >
        <div
          class="accordion__Header-l3iym9-5 kYfwMS"
        >
          Where can I watch?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-4 fLZyeo"
      >
        <div
          class="accordion__Header-l3iym9-5 kYfwMS"
        >
          How do I cancel?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-l3iym9-4 fLZyeo"
      >
        <div
          class="accordion__Header-l3iym9-5 kYfwMS"
        >
          What can I watch on Netflix?
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
